### PR TITLE
Add expandable error messages + more (Alt to PR #1)

### PR DESCRIPTION
**Pull request by hauntii**
This PR adds the following to Bandwidth:

- Collapsed errors that can be expanded by a button press, which auto-collapse after 45 seconds. ( They can be expanded at any time, even if the bot restarts. )

expandable-errors

- Changed the automatic member counter to be updated every five minutes, rather than on join, to prevent ratelimiting if too many accounts were to join / leave at the same time.
- Moved util.js into the new `utils` folder as `timed.js`.
- Refactored the messageCreate.js to split out the error code functions into a utility in the `utils` folder. This is done to simplify adding more events, along with allowing the expandable errors to function without a circular loop.

This is an alternate PR to PR #1, as they will conflict with each other.

### CODE REVIEW

**Comment by jonbarrow** [comment]
Gave it a glance before work, will do a more in-depth review later today

Looks really good so far! I like the collapsing a lot

My only 2 areas of concern right now are the error code embed color and the timers for them to auto collapse again

It feels a bit odd to have the embed color initially be the color for the WiiU errors, makes it seem like the error is going to be a WiiU error? Maybe swap it out for our general brand color, 0x131733, initially before it gets expanded to show its console color. Or have the initial color _be_ the consoles color

I also see that situations where someone has an expanded error and the bot restarts isn't handled. If someone expands the error, and the bot restarts for whatever reason, then their emebed will never collapse again. Usually to handle these situations I also store active timers in a database with their end times and any relevant data (like in this case the message ID), and then on boot going through them all and self-recovering. Though that might be a bit over kill here since we'd be storing potentially thousands of rows of the DB just for 45 second timers. What do you think?

**Comment by hauntii** [comment]
On the embed color, I've decided to switch it to the color for the said error, as that provides a better preview.

For the timer issues while rebooting, I don't think it's necessary to keep them saved as it isn't *too* important to have them forcefully collapse. It's also a pretty rare situation for that to occur, so I don't think it's enough to worry about.

**Comment by CirnoMoment** [comment]
Clearly a 1/10 bot, doesn't even have an owo command, the most important part of the bot, without it, it's nothing😤😤😤. Also maybe have a second button to say "remove error" or something like that, just in case of false positives, as we have seen before in several cases now.

**Comment by hauntii** [comment]
I've whipped up some less placeholder-y images for some of the embeds, and I'm unsure if I should just add these on to this PR, or create a new one. 

<details>
  <summary>Newer images</summary>
  
  accepted
  denied
  pending
  piracy

</details>

Which do you prefer I do, @jonbarrow ?

File: src/events/messageCreate.js [hunk 1]
```diff
@@ -1,15 +1,10 @@
 const Discord = require('discord.js');
 const { button: disableNLPButton } = require('../buttons/disable-nlp');
-const wiiuSupportCodes = require('../console-errors/wiiu/support-codes');
-const threeDSSupportCodes = require('../console-errors/3ds/support-codes');
-const pretendoSupportCodes = require('../console-errors/pretendo/support-codes');
+const { button: expandErrorButton } = require('../buttons/expand-error');
+const checkForErrorCode = require('../utils/errorCode').checkForErrorCode;
```
Comment (jonbarrow) [inline src/events/messageCreate.js:4]: For consistencies sake this should be written  as `const { checkForErrorCode } = require('../utils/errorCode');`

File: src/buttons/expand-error.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+const Discord = require('discord.js');
+const checkForErrorCode = require("../utils/errorCode").checkForErrorCode;
```
Comment (jonbarrow) [inline src/buttons/expand-error.js:2]: For consistencies sake this should be written as `const { checkForErrorCode } = require('../utils/errorCode');`

File: src/setup-guild.js [hunk 3]
```diff
@@ -1,11 +1,12 @@
 const Discord = require('discord.js');
 const { REST } = require('@discordjs/rest');
 const { Routes } = require('discord-api-types/v10');
-const util = require('./util');
+const timedUtils = require('./utils/timed');
```
Comment (jonbarrow) [inline src/setup-guild.js:4]: The `errorCode` utils function is imported directly and not the entire module, but `timed` imports the entire module and then uses the function with dot notation. This is inconsistent. Either `errorCode` needs to be imported as a whole module and use dot notation like here, or `timed` needs to import only the function with destructuring

**Review by jonbarrow — CHANGES_REQUESTED** [review]
Reviewed the PR, looks really good! Only a few small changes for more consistent code is all

**Comment by jonbarrow** [comment]
> Which do you prefer I do, @jonbarrow ?

I would say make a new PR for these new images. Thank you for that

**Comment by hauntii** [comment]
Should be all ready to merge :)

**Comment by jonbarrow** [comment]
Looks awesome! Thanks a bunch!

**Comment by jonbarrow** [comment]
Bot has been updated and deployed on the live server, you should now see these changes in action. I look forward to your next PR!

**Pull request merged**